scripts/run_backend_tests_test.py

```diff
@@ -0,0 +1,888 @@
+                tasks, task_to_taskspec, True)
+
+        self.assertIn(
+            'An unexpected error occurred. '
+            'Task output:\nXYZ',
+            self.print_arr)
+
+    def test_invalid_directory_in_sys_path_throws_error(self) -> None:
+        with self.swap_install_third_party_libs:
+            from scripts import run_backend_tests
+
+        def mock_path_exists(dirname: str) -> bool:
+            for directory in common.DIRS_TO_ADD_TO_SYS_PATH:
+                if os.path.dirname(directory) == dirname:
+                    return False
+            return True
+        swap_path_exists = self.swap(os.path, 'exists', mock_path_exists)
+
+        with swap_path_exists, self.assertRaisesRegex(
+            Exception,
+            'Directory %s does not exist.' % common.DIRS_TO_ADD_TO_SYS_PATH[0]
+        ):
+            run_backend_tests.main(args=[]) # type: ignore[no-untyped-call]
+
+    def test_invalid_delimiter_in_test_path_argument_throws_error(self) -> None:
+        with self.swap_install_third_party_libs:
+            from scripts import run_backend_tests
+
+        with self.swap_fix_third_party_imports, self.assertRaisesRegex(
+            Exception,
+            r'The delimiter in test_path should be a slash \(/\)'
+        ):
+            run_backend_tests.main( # type: ignore[no-untyped-call]
+                args=['--test_path', 'scripts.run_backend_tests'])
+
+    def test_invalid_delimiter_in_test_target_argument_throws_error(
+            self) -> None:
+        with self.swap_install_third_party_libs:
+            from scripts import run_backend_tests
+
+        with self.swap_fix_third_party_imports, self.assertRaisesRegex(
+            Exception,
+            r'The delimiter in test_target should be a dot \(\.\)'
+        ):
+            run_backend_tests.main( # type: ignore[no-untyped-call]
+                args=['--test_target', 'scripts/run_backend_tests'])
+
+    def test_invalid_test_target_message_is_displayed_correctly(self) -> None:
+        with self.swap_install_third_party_libs:
+            from scripts import run_backend_tests
+        swap_check_results = self.swap(
+            run_backend_tests, '_check_test_results',
+            lambda *unused_args, **unused_kwargs: (100, 0, 0, 0))
+        swap_check_coverage = self.swap(
+            run_backend_tests, '_check_coverage',
+            lambda *unused_args, **unused_kwargs: ('', 100.00))
+        with self.swap_fix_third_party_imports, self.swap_execute_task:
+            with swap_check_coverage, self.swap_redis_server, self.print_swap:
+                with self.swap_cloud_datastore_emulator, swap_check_results:
+                    run_backend_tests.main( # type: ignore[no-untyped-call]
+                        args=[
+                            '--test_target',
+                            'scripts.run_backend_tests.py'
```

Comment:
```suggestion
                            'scripts.run_backend_tests'
```

Do we automatically strip the `.py` if it's provided? Normally it shouldn't be included in the `test_target` argument, right?

Reply:
This is the test to check the same functionality. We replace `.py` with `_test` and display the warning msg `WARNING : test_target flag should point to the test file.` to the dev.